proxy/src/main/java/com/velocitypowered/proxy/plugin/omnibus/OmnibusEventManager.java

```diff
@@ -0,0 +1,208 @@
+package com.velocitypowered.proxy.plugin.omnibus;
+
+import com.velocitypowered.api.event.EventHandler;
+import com.velocitypowered.api.event.EventManager;
+import com.velocitypowered.api.event.PostOrder;
+import com.velocitypowered.api.event.Subscribe;
+import com.velocitypowered.api.plugin.PluginContainer;
+import com.velocitypowered.api.plugin.PluginManager;
+import org.apache.logging.log4j.LogManager;
+import space.arim.omnibus.events.AsyncEvent;
+import space.arim.omnibus.events.EventBus;
+import space.arim.omnibus.events.ListenerPriorities;
+import space.arim.omnibus.events.RegisteredListener;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+public class OmnibusEventManager implements EventManager {
```

Comment:
we're using omnibus for events now? cool :0

Reply:
Yep. There is far more purpose to it than just using Omnibus for the sake of it, however.

Omnibus will let us use asynchronous events. That means we'll block less threads and consume fewer system resources with asynchronous event resumption.

Async events have been added in a different form to Velocity 2.0, but Velocity 2.0 is a long way away.